Tests/SwiftDocCUtilitiesTests/PreviewActionIntegrationTests.swift

```diff
@@ -330,6 +462,42 @@ class PreviewActionIntegrationTests: XCTestCase {
     
     // MARK: -
     
+    private func createTemporaryTestSocketPath() throws -> String {
+        // The unix domain socket paths have a character limit that we need to stay under.
+        
+        func isShortEnoughUnixDomainSocket(_ url: URL) -> Bool {
+            url.path.utf8.count <= 103
+        }
+        
+        // Prefer a temporary socket URL that's relative to the unit test bundle location if possible
+        let bundleRelativeSocketURL = try createTemporaryDirectory().appendingPathComponent("s", isDirectory: false)
+        if isShortEnoughUnixDomainSocket(bundleRelativeSocketURL) {
+            return bundleRelativeSocketURL.path
+        }
+        
+        // If that URL was to long, try a temporary socket URL in the user's shared temporary directory.
+        // The added "doc" and UUID components should be sufficient to avoid collisions with other tests or other processes.
+        
+        let tempDir = URL(fileURLWithPath: Foundation.NSTemporaryDirectory())
+            .appendingPathComponent("docc", isDirectory: true)
+            .appendingPathComponent(UUID().uuidString, isDirectory: true)
+        
+        let socketURL = tempDir.appendingPathComponent("s", isDirectory: false)
+        
+        // If we couldn't create short enough socket path, skip the tests rather than fail them to avoid flakiness in the CI.
+        // The current implementation _should_ result in a path that's around 92 characters long, so the 11 character headroom
+        // should cover some amount of hypothetical changes to the `NSTemporaryDirectory` length and the `UUID().uuidString` length.
+        try XCTSkipIf(!isShortEnoughUnixDomainSocket(socketURL),
```

Comment:
This is an interesting XCTest feature: Skipping a test after it has started!

Reply:
Yes. It's typically done as the first step of a test but here I'm using it to stop before a detectable "failure" that's more caused by the environment in which the tests are executed than the test itself.